app/domain/nextdate.go

```diff
@@ -0,0 +1,46 @@
+package domain
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func GetNextDate(now time.Time, date string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", errors.New("repeat is empty string")
+	} else if strings.Contains(repeat, "d ") {
```

Comment:
Так как внтури каждого if есть выход, то можно не оборачивать в else
```suggestion
	} 
	if strings.Contains(repeat, "d ") {
```